contracts/truefi2/TrueLender2.sol

```diff
@@ -253,16 +295,46 @@ contract TrueLender2 is ITrueLender2, Claimable {
      * @param loanToken Loan to reclaim capital from
      * @param pool Pool from which the loan was funded
      */
-    function _redeemAndRepay(ILoanToken2 loanToken, ITrueFiPool2 pool) internal returns (uint256 fundsReclaimed) {
+    function _redeemAndRepay(
+        ILoanToken2 loanToken,
+        ITrueFiPool2 pool,
+        bytes calldata data
+    ) internal returns (uint256 fundsReclaimed) {
         // call redeem function on LoanToken
         uint256 balanceBefore = pool.token().balanceOf(address(this));
         loanToken.redeem(loanToken.balanceOf(address(this)));
         uint256 balanceAfter = pool.token().balanceOf(address(this));
 
         // gets reclaimed amount and pays back to pool
         fundsReclaimed = balanceAfter.sub(balanceBefore);
-        pool.token().approve(address(pool), fundsReclaimed);
-        pool.repay(fundsReclaimed);
+        uint256 fee = _swapFee(pool.token(), loanToken, data);
+
+        pool.token().approve(address(pool), fundsReclaimed.sub(fee));
```

Comment:
we need to deposit the fee into a feePool (tfUSDC), and pay the fee in that token. So after the swap deposit the token into the pool, transfer the pool tokens to the staking contract.